service/src/main/java/bio/terra/workspace/service/workspace/model/Workspace.java

```diff
@@ -191,4 +192,43 @@ public Workspace build() {
           workspaceStage);
     }
   }
+
+  /**
+   * If requester only has discoverer role, they can only see a subset of workspace. They cannot
+   * see:
+   *
+   * <ul>
+   *   <li>Workspace description
+   *   <li>Spend profile
+   *   <li>Can see special type, short description and version properties, and no other properties
+   * </ul>
+   */
+  public static Workspace stripWorkspaceForRequesterWithOnlyDiscovererRole(
+      Workspace fullWorkspace) {
+    Workspace.Builder strippedWorkspace =
+        new Builder()
+            .workspaceId(fullWorkspace.getWorkspaceId())
+            .userFacingId(fullWorkspace.getUserFacingId())
+            .workspaceStage(fullWorkspace.getWorkspaceStage())
+            .displayName(fullWorkspace.getDisplayName().orElse(""));
+
+    Map<String, String> strippedProperties = new HashMap<>();
+    if (fullWorkspace.getProperties().containsKey(Properties.TYPE)) {
+      strippedProperties.put(Properties.TYPE, fullWorkspace.getProperties().get(Properties.TYPE));
+    }
+    if (fullWorkspace.getProperties().containsKey(Properties.SHORT_DESCRIPTION)) {
+      strippedProperties.put(
+          Properties.SHORT_DESCRIPTION,
+          fullWorkspace.getProperties().get(Properties.SHORT_DESCRIPTION));
+    }
+    if (fullWorkspace.getProperties().containsKey(Properties.VERSION)) {
+      strippedProperties.put(
+          Properties.VERSION, fullWorkspace.getProperties().get(Properties.VERSION));
+    }
+    if (strippedProperties.size() > 0) {
```

Comment:
I don't think you need this `if`. Assigning an empty map is fine. If you don't assign it, the builder turns around and creates one.

Reply:
Done